src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -44,9 +44,9 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             {
                 // look for access token in the cache first.
                 // no access token is found, then it means token does not exist
-                // or new assertion has been passed. We should not use Refresh Token
-                // for the user because the new incoming token may have updated claims
-                // like MFA etc.
+                // or new assertion has been passed. 
+                // Look for a refresh token, if refresh token is found perform refresh token flow.
+                // If a refresh token is not found, fetch new access token for OBO
```

Comment:
How do we do that?  If we don't have a refresh token? I'm a bit confused? do you mean if the access token is not found, fetch a new access token, @neha-bhargava ?

Reply:
We land here when AT is not found in the cache. Earlier we did not look for RT and just fetched a fresh AT. Now, we look for RT in the cache with same UserAssertion as request and use it if it is found. If not found we fetch a fresh AT

Reply:
do we have a test for this:
- new UserAssertion(AT1) -> cache OBO_AT1/RT1 pair
- same UserAssertion(AT1) -> use cached OBO_RT1
- new UserAssertion(AT2 (just a claim change, same user)) -> cache OBO_AT2/RT2 pair
- same UserAssertion(AT2) -> cached OBO_RT2
- same UserAssertion(AT1) -> use cached OBO_RT1